test/lifecycle_module_test.dart

```diff
@@ -409,6 +436,60 @@ void runTests(bool runSpanTests) {
           await module.load();
         });
 
+        group('should record user specified timing', () {
+          DateTime startTime;
+
+          setUp(() async {
+            final Completer<DateTime> startTimeCompleter = new Completer();
+
+            subs.add(getTestTracer()
+                .onSpanFinish
+                .where(
+                    (span) => span.operationName == 'TestLifecycleModule.load')
+                .listen(expectAsync1((span) {
+              startTimeCompleter.complete(span.startTime);
+            })));
+
+            await module.load();
+
+            startTime = await startTimeCompleter.future;
+          });
+
+          tearDown(() {
+            startTime = null;
+          });
+
+          [
+            StartupTimingType.firstUseful,
+          ].forEach((specifier) {
+            test('should specify timing for ${specifier.operationName}',
+                () async {
+              subs.add(getTestTracer()
+                  .onSpanFinish
+                  .where((span) =>
+                      span.operationName ==
+                      'TestLifecycleModule.${specifier.operationName}')
+                  .listen(expectAsync1((span) {
+                expect(span.startTime, startTime);
+              })));
+              module.specifyStartupTiming(specifier);
```

Comment:
We should have test coverage for custom and default tags/references passed into this. (same for any shorthand methods we have).